rust/src/lib/route.rs

```diff
@@ -323,6 +331,17 @@ impl RouteEntry {
                 self.next_hop_addr = Some(new_via);
             }
         }
+        if let Some(src) = self.source.as_ref() {
+            let new_src = format!("{}", src.parse::<std::net::IpAddr>()?);
```

Comment:
Instead of trust auto error converter, I think it is better to `map_err` providing better error message.